sys/include/net/gnrc/ndp2.h

```diff
@@ -0,0 +1,355 @@
+/*
+ * Copyright (C) 2017 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    net_gnrc_ndp2   IPv6 Neighbor discovery (v2)
+ * @ingroup     net_gnrc_ipv6
+ * @brief       Provides build and send functions for neighbor discovery packets
+ * @{
+ *
+ * @file
+ * @brief       GNRC-specific neighbor discovery definitions
+ *
+ * @author  Martine Lenders <m.lenders@fu-berlin.de>
+ */
+#ifndef GNRC_NDP2_H
+#define GNRC_NDP2_H
+
+#include <stdint.h>
+
+#include "kernel_types.h"
+#include "net/gnrc/pkt.h"
+#include "net/gnrc/ipv6/netif.h"
+#include "net/ipv6/addr.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   @ref net_gnrc_nettype to send NDP packets to
+ */
+#ifndef GNRC_NETTYPE_NDP2
+# if    defined(MODULE_GNRC_IPV6) || DOXYGEN
+#  define GNRC_NETTYPE_NDP2 (GNRC_NETTYPE_IPV6)
```

Comment:
so, maybe mention this fact here? It's really confusing to bend things in the implementation that are oly needed for testing.